extension/source/QuillPage/components/OnboardingActionPanel.tsx

```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+
+import PasswordCreationPanel from './PasswordCreationPanel';
+import QuickColumn from './QuickColumn';
+import SetNicknamePanel from './SetNicknamePanel';
+import ViewSecretPhrasePanel from './ViewSecretPhrasePanel';
+import WorkflowNumbers from './WorkflowNumbers';
+
+const OnboardingActionPanel: React.FunctionComponent<{ pageIndex: number }> = ({
+  pageIndex,
+}) => (
+  <QuickColumn>
+    <WorkflowNumbers
+      current={pageIndex + 1}
+      max={3}
+      onSelect={(pageNumber) => {
+        // Note: This is for demo purposes only. We're not going to be using
+        // search parameters like this.
+        window.location.href = `?p=${pageNumber}`;
```

Comment:
I feel that ruins the separation though. `WorkflowNumbers` is an isolated thing that doesn't know anything specific about what's going on. `OnboardingActionPanel` on the other hand is clearly concerned with the specific use case.

Reply:
hmm, that makes sense if WorkflowNumbers is also used in other contexts than changing pages. Otherwise can just rename the component to be called a PageSwitcher and let it handle page switching.

Wonder what @jacque006 has to say.

Reply:
I would just [Rule of Three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) it. If we repeat this pattern multiple times makes sense to break it into a separate component.

Reply:
Done.

Reply:
Done.